src/main/java/pl/security_project/service/UserService.java

```diff
@@ -0,0 +1,42 @@
+package pl.security_project.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import pl.security_project.repository.UserRepository;
+import pl.security_project.repository.UserRoleRepository;
+import pl.security_project.user.User;
+import pl.security_project.user.UserRole;
+
+@Service
+public class UserService {
+
+    private static final String DEFAULT_ROLE = "ROLE_USER";
+    private UserRepository userRepository;
+    private UserRoleRepository roleRepository;
+    private PasswordEncoder passwordEncoder;
+
+    @Autowired
+    public UserService(PasswordEncoder passwordEncoder) {
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    @Autowired
+    public void setUserRepository(UserRepository userRepository) {
```

Comment:
This is weird. You could inject userRepo with constructor.

Reply:
What is the difference ?